executor/instruction/has_executor.rs

```diff
@@ -99,10 +99,9 @@ impl HasExecutor {
         let owner = has.owner().as_variable().unwrap();
         let attribute = has.attribute().as_variable().unwrap();
 
-        let output_tuple_positions = if iterate_mode.is_inverted() {
-            TuplePositions::Pair([Some(attribute), Some(owner)])
-        } else {
-            TuplePositions::Pair([Some(owner), Some(attribute)])
+        let output_tuple_positions = match iterate_mode {
+            BinaryIterateMode::Unbound => TuplePositions::Pair([Some(owner), Some(attribute)]),
+            _ => TuplePositions::Pair([Some(attribute), Some(owner)]),
```

Comment:
For the `HasExecutor` (rest. the `HasReverseExectutor), what exactly does the TuplePosition convey? 
I think I heard you say the sort variable is always in the first position? Does it tell us how to parse the two parts of the key?

Reply:
This associates the positions of the variable values in the output of the tuple iterator with their positions in the output row.